src/frontend/servers/y-provider/__tests__/server.test.ts

```diff
@@ -0,0 +1,230 @@
+} from '@hocuspocus/provider';
+import request from 'supertest';
+import WebSocket from 'ws';
+
+const port = 5555;
+const portWS = 6666;
+const origin = 'http://localhost:3000';
+
+jest.mock('../src/env', () => {
+  return {
+    PORT: port,
+    COLLABORATION_SERVER_ORIGIN: origin,
+    COLLABORATION_SERVER_SECRET: 'test-secret-api-key',
+  };
+});
+
+console.error = jest.fn();
+
+import { hocuspocusServer, initServer } from '../src/server'; // Adjust the path to your server file
+
+const { app, server } = initServer();
+
+describe('Server Tests', () => {
+  beforeAll(async () => {
+    await new Promise<void>((resolve) => {
+      if (server.listening) {
+        resolve();
+      }
+    });
+
+    await new Promise<void>((resolve) => {
+      void hocuspocusServer
+        .configure({
+          port: portWS,
+        })
+        .listen()
+        .then(() => {
+          resolve();
+        });
+    });
+  });
+
+  afterAll(() => {
+    server.close();
+    void hocuspocusServer.destroy();
+  });
+
+  test('Ping Pong', async () => {
+    const response = await request(app as any).get('/ping');
+
+    expect(response.status).toBe(200);
+    expect(response.body.message).toBe('pong');
+  });
+
+  test('POST /collaboration/api/reset-connections?room=[ROOM_ID] invalid origin', async () => {
+    const response = await request(app as any)
+      .post('/collaboration/api/reset-connections/?room=test-room')
+      .set('Origin', 'http://invalid-origin.com')
+      .send({ document_id: 'test-document' });
+
+    expect(response.status).toBe(403);
+    expect(response.body.error).toBe('CORS policy violation: Invalid Origin');
+  });
+
+  test('POST /collaboration/api/reset-connections?room=[ROOM_ID] with incorrect API key should return 403', async () => {
+    const response = await request(app as any)
+      .post('/collaboration/api/reset-connections/?room=test-room')
+      .set('Origin', origin)
+      .set('Authorization', 'wrong-api-key');
+
+    expect(response.status).toBe(403);
+    expect(response.body.error).toBe('Forbidden: Invalid API Key');
+  });
+
+  test('POST /collaboration/api/reset-connections?room=[ROOM_ID] failed if room not indicated', async () => {
+    const response = await request(app as any)
+      .post('/collaboration/api/reset-connections/')
+      .set('Origin', origin)
+      .set('Authorization', 'test-secret-api-key')
+      .send({ document_id: 'test-document' });
+
+    expect(response.status).toBe(400);
+    expect(response.body.error).toBe('Room name not provided');
+  });
+
+  test('POST /collaboration/api/reset-connections?room=[ROOM_ID] with correct API key should reset connections', async () => {
+    const response = await request(app as any)
+      .post('/collaboration/api/reset-connections?room=test-room')
+      .set('Origin', origin)
+      .set('Authorization', 'test-secret-api-key');
+
+    expect(response.status).toBe(200);
+    expect(response.body.message).toBe('Connections reset');
```

Comment:
do you want to test whether the connection has actually been reset (i.e.: users "kicked"?

Reply:
I added a check on `hocuspocusServer.closeConnections`, I don't succeed to do a full check test, I think we are in half mock env with Jest, so users don't really connect totally to the websocket, or they are kicked out because of a ping pong thing between the HocusPocus server and the HocusPocus Provider?

I have as well a e2e test on that assert the deconnection and the reconnection. I will try to came back on this tests when I will get more times.